pydantic/main.py

```diff
@@ -173,20 +174,36 @@ def prepare_field(cls, field: 'ModelField') -> None:
         pass
 
 
-def inherit_config(self_config: 'ConfigType', parent_config: 'ConfigType', **namespace: Any) -> 'ConfigType':
-    if not self_config:
-        base_classes: Tuple['ConfigType', ...] = (parent_config,)
-    elif self_config == parent_config:
-        base_classes = (self_config,)
-    else:
-        base_classes = self_config, parent_config
+def inherit_config(
+    bases: Collection[Type[object]], config_from_namespace: Type[object], config_namespace: Dict[str, Any]
```

Comment:
```suggestion
    bases: Collection[Type[object]], config_from_namespace: Type[object], config_namespace: Dict[str, Any]
```
```suggestion
    bases: Sequence[Type[object]], config_from_namespace: Type[object], config_namespace: Dict[str, Any]
```

Reply:
You're right: bases order is the most important.